addons/html_editor/static/src/utils/formatting.js

```diff
@@ -90,7 +97,10 @@ export const formatsSpecs = {
         removeStyle: (node) => removeStyle(node, "font-size"),
     },
     setFontSizeClassName: {
-        isFormatted: (node) => FONT_SIZE_CLASSES.find((cls) => node?.classList?.contains(cls)),
+        isFormatted: (node) => {
+            const element = isTextNode(node) ? closestElement(node) : node;
+            return FONT_SIZE_CLASSES.find((cls) => element?.classList.contains(cls));
+        },
```

Comment:
Why check `isTextNode` to change it to `closestElement`? `closestElement` of a element node is the element node itself.

This shall be enough
```suggestion
        isFormatted: (node) => FONT_SIZE_CLASSES.find((cls) => closestElement(node)?.classList?.contains(cls)),
```

Also the current fix won't work if node has `font-size` style rather than classes. You will have to change the `isFormatted` in `fontSize` too. Also if you could add a test case that.